From edb4e584d384ddaa8b433654fcd7b29c44e060ac Mon Sep 17 00:00:00 2001
From: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
Date: Thu, 13 Oct 2016 16:36:43 +0200
Subject: [PATCH] Set ENET_REF_CLK to 50MHz if phy-mode rmii

If fec device tree property 'phy-mode' set to 'rmii' then set ENET_REF_CLK to
50MHz.

Signed-off-by: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
---
 arch/arm/mach-imx/clk-imx6q.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/clk-imx6q.c b/arch/arm/mach-imx/clk-imx6q.c
index bc3feec..dbd33c8 100644
--- a/arch/arm/mach-imx/clk-imx6q.c
+++ b/arch/arm/mach-imx/clk-imx6q.c
@@ -19,6 +19,8 @@
 #include <linux/of.h>
 #include <linux/of_address.h>
 #include <linux/of_irq.h>
+#include <linux/of_net.h>
+#include <linux/phy.h>
 #include <dt-bindings/clock/imx6qdl-clock.h>
 
 #include "clk.h"
@@ -888,7 +890,15 @@ static void __init imx6q_clocks_init(struct device_node *ccm_node)
 
 	/*Set enet_ref clock to 125M to supply for RGMII tx_clk */
 	clk_set_rate(clk[IMX6QDL_CLK_ENET_REF], 125000000);
-
+	np = of_find_compatible_node(NULL, NULL, "fsl,imx6q-fec");
+	if (np) {
+		int ret;
+		ret = of_get_phy_mode(np);
+		if(ret >= 0 && ret == PHY_INTERFACE_MODE_RMII) {
+			printk("fec pyh-mode is 'rmii'. Set ENET_REF to 50MHz\n");
+			clk_set_rate(clk[IMX6QDL_CLK_ENET_REF], 50000000);
+		}
+	}
 #ifdef CONFIG_MX6_VPU_352M
 	/*
 	 * If VPU 352M is enabled, then PLL2_PDF2 need to be
-- 
2.6.6

